cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -250,7 +260,52 @@ public void getAllApps(HttpRequest request, HttpResponder responder,
       .map(ApplicationRecord::new)
       .collect(Collectors.toList());
 
-    responder.sendJson(HttpResponseStatus.OK, GSON.toJson(apps));
+    if (Optional.ofNullable(pageSize).orElse(0) != 0) {
+      PaginatedApplicationRecords result = getPaginatedResults(apps, pageToken, pageSize, orderBy,
+                                                               nameFilter);
+      responder.sendJson(HttpResponseStatus.OK, GSON.toJson(result));
+    } else {
+      responder.sendJson(HttpResponseStatus.OK, GSON.toJson(apps));
+    }
+  }
+
+  /**
+   * Returns paginated result for getApps, also includes nextPageToken
```

Comment:
Please add a TODO referencing your JIRA that says it's a temporary implementation